src/scripts.js

```diff
@@ -35,37 +37,64 @@ function onStartup() {
       cookbook = new Cookbook(values[1].recipeData, values[0].ingredientsData)
+      // console.log(cookbook)
       currentUser = new User(values[2])
       domUpdates.greetUser(currentUser)
       domUpdates.populateCards(cookbook.recipes, cardArea)
+      userPantry = new Pantry(currentUser)  
     })
-}
+  }
 
 function addToCookLater(event) {
-  let targetRecipe = cookbook.recipes.find(recipe => recipe.id === Number(event.target.id));
+  let targetRecipe = cookbook.recipes.find(recipe => recipe.id === +event.target.id);
   let cookingStatus = domUpdates.toggleNameClass(event, recipesToCookButton, 'cook-active')
   cookingStatus ? currentUser.addToFavorites('recipesToCook', targetRecipe) : currentUser.removeFromFavorites('recipesToCook', targetRecipe);
   console.log(currentUser.recipesToCook)
- }
+}
 
- function findRecipesToCook(){
-   const recipeIds = currentUser.recipesToCook.map(recipe => recipe.id)
-   domUpdates.viewSelectedCards(cardArea, currentUser, recipesToCookButton, recipeIds, 'recipesToCook');
- }
+function findRecipesToCook() {
+  const recipeIds = currentUser.recipesToCook.map(recipe => recipe.id)
+  domUpdates.viewSelectedCards(cardArea, currentUser, recipesToCookButton, recipeIds, 'recipesToCook');
+}
 
 function favoriteCard(event) {
   let specificRecipe = cookbook.recipes.find(recipe =>recipe.id  === +event.target.id);
   let favoriteStatus = domUpdates.toggleNameClass(event, favButton, 'favorite-active')
   favoriteStatus ? currentUser.addToFavorites('favoriteRecipes', specificRecipe) : currentUser.removeFromFavorites('favoriteRecipes', specificRecipe);
-  // console.log(currentUser.favoriteRecipes)
+}
+
+// start here
+
+function declareRecipe(event) {
+  let newRecipeInfo = cookbook.recipes.find(recipe => recipe.id === +event.target.id)
+  const currentRecipe = new Recipe(newRecipeInfo, ingredientsData);
+  // postIngredients(currentRecipe)
+  return currentRecipe
 }
 
 function findDirections(event) {
-  let newRecipeInfo = cookbook.recipes.find(recipe => recipe.id === Number(event.target.id))
-  let recipeObject = new Recipe(newRecipeInfo, ingredientsData);
-  let cost = recipeObject.calculateCost()
+  // let newRecipeInfo = cookbook.recipes.find(recipe => recipe.id === +event.target.id)
+  // let currentRecipe = new Recipe(newRecipeInfo, ingredientsData);
+  const currentRecipe = declareRecipe(event)
+  let cost = currentRecipe.calculateCost()
   let costInDollars = (cost / 100).toFixed(2)
-  domUpdates.displayDirections(cardArea, recipeObject, costInDollars)
+  domUpdates.displayDirections(cardArea, currentRecipe, costInDollars)
+  // postIngredients(currentRecipe)
+}
+
+function postIngredients(currentRecipe) {
+  let ingredientToUpdate = userPantry.ingredientsToPantry(currentRecipe)
+  let promisesReturned = ingredientToUpdate.reduce((returnValues, ingredient) => {
+    returnValues.push(User.updateUserPantry(ingredient))
+    return returnValues
+  },[])
+  return Promise.all(promisesReturned)
```

Comment:
here is the biggest change of all, since I could save the returned promise into a global variable, now this `postIngredients` function is returning the initial promise, down below in the click handler function, I invoke the function and the start chaining the `.then` methods

Reply:
Looks good, this definitely keeps things in better accordance of SRP.